### DISCUSSION TITLE
"Email has already been taken" after delete (or destroy ?) an user

### DISCUSSION THREAD

**Discussion by joyboy67** [solidusio/solidus#4643] [Troubleshooting]
Hello, when I delete/destroy an user without orders from admin panel (https://github.com/solidusio/solidus/discussions/4590) and try to register a new user with the same email I get the error "Email has already been taken" whereas the user no longer exists in `Spree::User.all`

**Env**
Rails version: 7.0.4
solidus_auth_devise (2.5.4)
solidus (3.2.2)

EDIT: 

I found with `Spree::User.with_discarded` that the users was just discarded, but the data should not be scramble with `scramble_email_and_password` ? (https://github.com/solidusio/solidus_auth_devise/blob/master/app/models/spree/user.rb)

**Comment by jarednorman** [comment]
Yeah, that does seem like a bug. I wonder if the guard clause in that method doesn't work correctly on newer versions of ActiveRecord or something.

**Reply by joyboy67** [reply]
Ah yes, may be, I use Rails 7.0.4
